ledger-tool/src/main.rs

```diff
@@ -4141,9 +4148,33 @@ fn main() {
                     force_update_to_open,
                 );
                 let num_slots = value_t_or_exit!(arg_matches, "num_slots", usize);
-                let slots = blockstore
-                    .get_latest_optimistic_slots(num_slots)
-                    .expect("Failed to get latest optimistic slots");
+                let include_vote_only_slots = arg_matches.is_present("include_vote_only_slots");
+
+                let slots_iter = blockstore
+                    .reversed_optimistic_slots_iterator()
+                    .expect("Failed to get reversed optimistic slots iterator");
+                let slots: Vec<_> = if include_vote_only_slots {
+                    // All slots ok so take num_slots items from the iterator
+                    slots_iter.take(num_slots).collect()
+                } else {
+                    // Otherwise, only include slots with at least one
+                    // transaction that is not the vote program
+                    slots_iter
+                        .filter(|(slot, _, _)| {
+                            let (entries, _, _) = blockstore
+                                .get_slot_entries_with_shred_info(*slot, 0, false)
+                                .expect("Failed to get slot entries");
+                            let contains_nonvotes = entries
+                                .iter()
+                                .flat_map(|entry| entry.transactions.iter())
+                                .flat_map(get_program_ids)
+                                .any(|program_id| *program_id != solana_vote_program::id());
+                            contains_nonvotes
+                        })
+                        .take(num_slots)
+                        .collect()
+                };
+
```

Comment:
for a follow-up, might make sense to indicate vote-only/economic when `--include-vote-only-slots` is passed

Reply:
Ahh yeah, I could see that being useful. I'm cool incorporating that into this change